server_test.go

```diff
@@ -107,7 +107,11 @@ var _ = Describe("Server", func() {
 			It("invokes the tool", func() {
 				stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"tools/call","params":{"name":"sha256sum","arguments":{"text":"the rain in spain falls mainly on the plains"}}}`)
 				Eventually(session.Out).Should(gbytes.Say("\n"))
-				Expect(lastResponse(session.Out.Contents())).To(MatchJSON(`{"jsonrpc":"2.0","id":2,"result":{"content":[{"type":"text","text":"b65aacbdd951ff4cd8acef585d482ca4baef81fa0e32132b842fddca3b5590e9"}],"isError":false}}`))
+
+				responses := allResponses(session.Out.Contents())
+				Expect(responses).To(HaveLen(4)) // First two are initialization, last two are the tool response
+				Expect(responses[len(responses)-2]).To(MatchJSON(`{"jsonrpc":"2.0","method":"test/notification","params":{"message":"Processing text"}}`))
```

Comment:
Ideally we might be able to make ordered assertions on the responses rather than having to refer to them by index.

gbytes.Say has an [implicit cursor](https://onsi.github.io/gomega/#codegbytescode-testing-streaming-buffers) that it moves forward as assertions pass but I don't think we can combine that easily with the MatchJSON matcher so this may be the best we can do.